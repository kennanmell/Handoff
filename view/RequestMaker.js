/**
 * Request Maker view class, which allows organizations
 * to make submit requests
 */


import React, { Component, PropTypes } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  TextInput,
  View,
  TouchableNativeFeedback,
  Modal,
  TouchableHighlight,
  Navigator,
  Alert
} from 'react-native';
import{Button} from 'native-base';

/*Builds a RequestMaker, which an Organization User uses to
submit requests. Has text boxes for the request title, description,
and tags. Acts as initial page for orgs. Has a button to submit requests,
edit org profile, edit previous requests and logout.*/
export default class RequestMaker extends Component {
    static propTypes = {
      onLogout:PropTypes.func.isRequired,
      onEditRequest:PropTypes.func.isRequired,
      onEditProfile:PropTypes.func.isRequired
    }
  constructor(props) {
    super(props);
    this.state = {text: '',
				  requestName: '',
				  requestDescription: '',
				  requestTags: ''
				  };
  }

  render() {
    return (

      <View style={{padding: 10}}>
      <Text style={styles.welcome}>
                    Handoff
                  </Text>
        <TextInput
          style={{height: 40}}
          placeholder="Give your request a snappy title"
          onChangeText={(text) => this.setState({requestName: text})}
          tag='titleInput'
        />
         <TextInput
                  style={{height: 40}}
                  placeholder="Provide details on what you are requesting"
                  onChangeText={(text) => this.setState({requestDescription: text})}
                  tag='detailsInput'
                />
         <TextInput
                  style={{height: 40}}
                  placeholder="Tag your request with keywords so people can find it"
                  tag='keywordsInput'
                  onChangeText={(text) => this.setState({requestTags: text})}
                />
        <Button
                    style={styles.button}
					onPress={() => {Alert.alert('Request Successful', 'Your request was successfully posted.');
					makeRequest(this.state.requestName, this.state.requestDescription);
					console.log("attempting to make request"); }}
                    tag='submitButton'
                    >Submit Request</Button>
        <Button
                    style={styles.button}
                    onPress={this.props.onEditProfile}
                    >Edit Profile</Button>
        <Button
                    style={styles.button}
                    onPress={this.props.onEditRequest}
                    >Edit Requests</Button>
        <Button
                  style={styles.button}
                  tag='logoutButton'
                  onPress={this.props.onLogout}
                  >Logout</Button>
      </View>
    );
  }
}

/* Given a requestName and a requestDescription, creates a request in the database.
For the moment, the request's organization and its tags are not modifiable. This functionality
will be added later.*/
async function makeRequest(requestName, requestDescription) {
    fetch('https://u116vqy0l2.execute-api.us-west-2.amazonaws.com/prod/requests/new', {
      method: 'POST',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
            "organization": "Hope Shelter",
			"time": 50,
			"title": requestName,
	        "description": requestDescription,
			"tags": ["food","test"]
      })
    })
	console.log("fetch maybe did something");
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },

    button: {
          flexDirection: 'column',
          justifyContent: 'center',
          alignItems: 'flex-end',
          marginLeft: 3,
          marginTop: 10,
          borderWidth: 1,
          padding: 5,
          borderColor: 'black',
          marginBottom: 15,
          backgroundColor: '#642D64',
      }
});