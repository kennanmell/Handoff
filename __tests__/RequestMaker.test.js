import 'react-native';
import React from 'react';
import RequestMaker from '../View/RequestMaker';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    	<RequestMaker
    		onLogout={ () => {
    		}}
    		
    		onEditRequest={ () => {
    		}}
    		
    		onEditProfile={ () => {
    		}}
    	/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});