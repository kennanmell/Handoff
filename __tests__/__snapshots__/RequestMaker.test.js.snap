exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
    Handoff
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Give your request a snappy title"
    style={
      Object {
        "height": 40,
      }
    } />
  <TextInput
    onChangeText={[Function]}
    placeholder="Provide details on what you are requesting"
    style={
      Object {
        "height": 40,
      }
    } />
  <TextInput
    onChangeText={[Function]}
    placeholder="Tag your request with keywords so people can find it"
    style={
      Object {
        "height": 40,
      }
    } />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }>
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }>
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }>
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }>
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;
